From 462715e408cbd64ed67930f472b487e52eb7fd5a Mon Sep 17 00:00:00 2001
From: Marvin Ramin <marvinramin@google.com>
Date: Wed, 16 Oct 2024 12:26:02 +0200
Subject: [PATCH] RESTRICT AUTOMERGE Use getLaunchedFromPackage instead of
 getCallingPackage

Ensure that MediaProjectionPermissionActivity is doing proper
attribution and validation of permissions before showing.

Bug: 373581993
Test: presubmit
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:7ea5bb592ed70fe57414f9bfc471bfbb6cc8735c)
Merged-In: Ibcc14090ce38a0fac21a394d861d044b9669a8a0
Change-Id: Ibcc14090ce38a0fac21a394d861d044b9669a8a0
---
 .../systemui/media/MediaProjectionPermissionActivity.java  | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/media/MediaProjectionPermissionActivity.java b/packages/SystemUI/src/com/android/systemui/media/MediaProjectionPermissionActivity.java
index 66c51d278dab..b016eff93028 100644
--- a/packages/SystemUI/src/com/android/systemui/media/MediaProjectionPermissionActivity.java
+++ b/packages/SystemUI/src/com/android/systemui/media/MediaProjectionPermissionActivity.java
@@ -62,11 +62,14 @@ public class MediaProjectionPermissionActivity extends Activity
     public void onCreate(Bundle icicle) {
         super.onCreate(icicle);
 
-        mPackageName = getCallingPackage();
+        // The source of truth where the original request is coming from.
+        mPackageName = getLaunchedFromPackage();
         IBinder b = ServiceManager.getService(MEDIA_PROJECTION_SERVICE);
         mService = IMediaProjectionManager.Stub.asInterface(b);
 
-        if (mPackageName == null) {
+        if (getCallingPackage() == null) {
+            // If get calling package is null, the activity was not started for result.
+            // We should not proceed.
             finish();
             return;
         }
-- 
2.47.1.613.gc27f4b7a9f-goog

